util/argo/diff_normalizer.go

```diff
@@ -0,0 +1,113 @@
+package argo
+
+import (
+	"encoding/json"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	jsonpatch "github.com/evanphx/json-patch"
+
+	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/util/diff"
+	"github.com/argoproj/argo-cd/util/settings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type normalizerPatch struct {
+	groupKind schema.GroupKind
+	namespace string
+	name      string
+	patch     jsonpatch.Patch
+}
+
+type normalizer struct {
+	patches []normalizerPatch
+}
+
+type overrideIgnoreDiff struct {
+	JsonPointers []string `yaml:"jsonPointers"`
+}
+
+// NewDiffNormalizer creates diff normalizer which removes ignored fields according to given application spec and resource overrides
+func NewDiffNormalizer(ignore []v1alpha1.ResourceIgnoreDifferences, overrides map[string]settings.ResourceOverride) (diff.Normalizer, error) {
+	for key, override := range overrides {
+		parts := strings.Split(key, "/")
+		if len(parts) < 2 {
+			continue
+		}
+		group := parts[0]
+		kind := parts[1]
+		if override.IgnoreDifferences != "" {
+			ignoreSettings := overrideIgnoreDiff{}
+			err := yaml.Unmarshal([]byte(override.IgnoreDifferences), &ignoreSettings)
+			if err != nil {
+				return nil, err
+			}
+
+			ignore = append(ignore, v1alpha1.ResourceIgnoreDifferences{
```

Comment:
Should we use a copy of ignore instead of modifying the original struct? I don't think this is an issue now, but it could be if we use the original struct again (or the parent struct).